userspace/libsinsp/utils.cpp

```diff
@@ -479,9 +479,9 @@ bool sinsp_utils::sockinfo_to_str(sinsp_sockinfo* sinfo, scap_fd_type stype, cha
 			sinfo->m_ipv4info.m_fields.m_l4proto == SCAP_L4_UDP)
 		{
 			ipv4tuple addr;
-			addr.m_fields.m_sip = *(uint32_t*)sb;
+			addr.m_fields.m_sip = sinfo->m_ipv4info.m_fields.m_sip;
 			addr.m_fields.m_sport = sinfo->m_ipv4info.m_fields.m_sport;
-			addr.m_fields.m_dip = *(uint32_t*)db;
+			addr.m_fields.m_dip = sinfo->m_ipv4info.m_fields.m_dip;
```

Comment:
Not sure why m_sport and m_dport don't need a memcpy (here and in line 529 or so)

Reply:
It looks like the source and destination fields are the same length. `memcpy`ing there would be pointless I think. Although later (on line 499) those fields are accessed with an array, do you think we need to `memcpy()` there?

Reply:
Oh wait. These are just copying from one struct to another. This is fine.

I can't really follow what's going on in the other chunk ATM (what is the source: an actual well behaved C struct or a packed struct at an arbitrary address in the event), so here's a general comment:

memcpy instead of an assignment is not about the size (it obviously must be the same for the source and destination) but about alignment: the destination is aligned properly for the type (simply because it *is* of that type) but if the source is a packed struct (or a random char* ptr) then technically treating it as a pointer to anything other than `char` is UB.

(AFAIK that also includes taking the address of a misaligned member in a packed struct)